### DISCUSSION TITLE
Adding items to atom array

### DISCUSSION THREAD

**Discussion by jukkahuuskonen** [facebookexperimental/Recoil#982] [Q&A]
Hi,

I'm very new to Recoil and I've been trying to solve this, but not managed to do that.

I got 2 atoms here like this:

```
export const items = atom({
  key: 'items',
  default: [],
});

export const newItems = atom({
  key: 'newItems ',
  default: [],
});
```

Now when `newItems` get updated, I want that array appended to the `items`. I can make it happen with hooks, but can I make it happen with recoil directly, maybe with selectors? Would simplify things considerably for me.

eg.
```
items === [], newItems === [1, 2]
-(magic happening)-> items === [1, 2]
newItems === [5]
-(magic happening)-> items === [1, 2, 5]
```
-Jukka

**Comment by BenjaBobs** [comment]
I think you can use `useRecoilCallback` to solve this use case.

**Comment by jukkahuuskonen** [comment]
> I think you can use `useRecoilCallback` to solve this use case.

OK, thanks, missed that one. I have to try and understand how it works :)

**Comment by jukkahuuskonen** [comment]
Couldn't get that `useRecoilCallback` to work the way I wanted so went to bed and after a good nights sleep (=restlessly turning back and forth thinking this...) I realized I don't actually need the `newItems`-atom itself at all. Instead, I just created a selector, `appendArrayToItems`, that gets the current `items` value, creates a new array from that and `newItems`-parameter and sets the `items` atom with it. Seems to work nicely.
And if `newItems´ atom was needed, I could set that also with that `appendArrayToItems` selector, but in my case, that is not needed.

```
export const items = atom({
  key: 'items',
  default: [],
});

export const appendArrayToItems = selector({
  key: 'appendArrayToItems',
  set: ({ get, set }, newItems = []) => {
    if (newValue.length > 0) {
      const currentItems = get(items);
      const appendedItems = [...currentItems, ...newItems];
      set(items, appendedItems);
    }
  },
});
```

**Comment by christopanayotovvia** [comment]
You can use [useSetRecoilState](https://recoiljs.org/docs/api-reference/core/useSetRecoilState) and make use of 

```
const setMessages = useSetRecoilState(messages);
 setMessages((currentState) => [
      ...currentState,
      newMessage,
    ]);


```
